### PR TITLE
Add environment variable to control logging of variables and gradients

### PR DESCRIPTION
**Pull request by reuben**
I had to do a bit of refactoring to make the code that extracts the results from the session.run call easier to read.

### CODE REVIEW

File: DeepSpeech.ipynb [hunk 1]
```diff
@@ -1535,11 +1538,14 @@
     "    batches_per_device = ceil(float(data_set.total_batches) / len(available_devices))\n",
     "\n",
     "    total_loss = 0.0\n",
-    "    params = [avg_loss]\n",
+    "    params = OrderedDict()\n",
```
Comment (tilmankamp) [inline DeepSpeech.ipynb]: Nice one! Cleaner than my raw list index approach.

File: DeepSpeech.ipynb [hunk 2]
```diff
@@ -1387,7 +1389,8 @@
     "            avg_tower_gradients = average_gradients(tower_results[1])\n",
     "\n",
     "            # Add logging of averaged gradients\n",
-    "            log_grads_and_vars(avg_tower_gradients)\n",
+    "            if log_variables:\n",
+    "                log_grads_and_vars(avg_tower_gradients)\n",
```
Comment (tilmankamp) [inline DeepSpeech.ipynb]: Are we sure that this is not touching the model layout? It would be enough to just exclude the merged summaries retrieval from `session.run`. However: If this is not the case, I am fine with it.

File: DeepSpeech.ipynb [hunk 3]
```diff
@@ -176,7 +177,8 @@
     "log_device_placement = bool(int(os.environ.get('ds_log_device_placement', 0)))\n",
     "random_seed = int(os.environ.get('ds_random_seed', 4567)) # To be adjusted in case of bad luck\n",
     "for var in ['b1', 'h1', 'b2', 'h2', 'b3', 'h3', 'b5', 'h5', 'b6', 'h6']:\n",
-    "    locals()['%s_stddev' % var] = float(os.environ.get('ds_%s_stddev' % var, default_stddev))"
+    "    locals()['%s_stddev' % var] = float(os.environ.get('ds_%s_stddev' % var, default_stddev))\n",
+    "log_variables = bool(len(os.environ.get('ds_log_variables', '')))"
```
Comment (tilmankamp) [inline DeepSpeech.ipynb]: You missed the documentation in the upper markdown part.

File: DeepSpeech.ipynb (hunk 2 above)
Comment (reuben) [inline DeepSpeech.ipynb]: It does change the graph, as we're not adding the summary ops anymore, but I don't think it would break checkpointing or something like that. I'll test it.

**Comment by reuben** [comment]
I ran the network on LapsBM for 10 epochs on the CPU with logging enabled and disabled. Looks like the same 25% speedup seen by @lissyx. The variance in the first epochs is probably due to me using the computer while it's training.

Plot

File: DeepSpeech.ipynb (hunk 2 above)
Comment (reuben) [inline DeepSpeech.ipynb]: Tested, restoring the checkpoint works normally with and without the ops.

**Comment by tilmankamp** [comment]
@reuben Looks great and promising. Please add the missing documentation and I'll approve.

**Comment by reuben** [comment]
I forgot to follow up on your comment. Should be ready to merge now.
